### ISSUE TITLE
collector fails after upgrading from 1.39.9 to 1.40.2

### ISSUE THREAD

**Issue by lycplus** [openzipkin/zipkin#1127]
It throws:

```
ERROR: unhandled error processing traceId -> spanId: .....
com.datastax.driver.core.exceptions.InvalidQueryException: Partition KEY part service_name cannot be restricted by IN relation (only the last part of the partition key can)
    at com.datastax.driver.core.exceptions.InvalidQueryException.copy(InvalidQueryException.java:50)
    at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
    at com.datastax.driver.core.AbstractSession.prepare(AbstractSession.java:110)
    at org.twitter.zipkin.storage.cassandra.Repository.<init>(Repository.java:170)
    at com.twitter.zipkin.cassandra.CassandraSpanStoreFactory$class.com$twitter$zipkin$cassandra$CassandraSpanStoreFactory$$lazyRepository(CassandraSpanStoreFactory.scala:48)
    at Evaluator__7f6b7f0a6004614ba97f0dced7c56242f559654a_2092793443$Factory$2$.com$twitter$zipkin$cassandra$CassandraSpanStoreFactory$$lazyRepository$lzycompute((inline):33)
    at Evaluator__7f6b7f0a6004614ba97f0dced7c56242f559654a_2092793443$Factory$2$.com$twitter$zipkin$cassandra$CassandraSpanStoreFactory$$lazyRepository((inline):33)
    at com.twitter.zipkin.cassandra.CassandraSpanStoreFactory$$anon$1.repository$lzycompute(CassandraSpanStoreFactory.scala:52)
    at com.twitter.zipkin.cassandra.CassandraSpanStoreFactory$$anon$1.repository(CassandraSpanStoreFactory.scala:52)
    at com.twitter.zipkin.storage.cassandra.CassandraSpanStore$$anonfun$apply$10.apply(CassandraSpanStore.scala:233)
    at com.twitter.zipkin.storage.cassandra.CassandraSpanStore$$anonfun$apply$10.apply(CassandraSpanStore.scala:228)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
    at scala.collection.immutable.List.foreach(List.scala:381)
    at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
    at scala.collection.immutable.List.map(List.scala:285)
    at com.twitter.zipkin.storage.cassandra.CassandraSpanStore.apply(CassandraSpanStore.scala:228)
    at com.twitter.zipkin.collector.builder.CollectorServiceBuilder$$anonfun$apply$1$$anonfun$1.apply(CollectorServiceBuilder.scala:65)
    at com.twitter.zipkin.collector.builder.CollectorServiceBuilder$$anonfun$apply$1$$anonfun$1.apply(CollectorServiceBuilder.scala:64)
    at com.twitter.zipkin.receiver.kafka.KafkaStreamProcessor$$anonfun$run$1.apply(KafkaStreamProcessor.scala:21)
    at com.twitter.zipkin.receiver.kafka.KafkaStreamProcessor$$anonfun$run$1.apply(KafkaStreamProcessor.scala:17)
    at scala.collection.Iterator$class.foreach(Iterator.scala:742)
    at kafka.utils.IteratorTemplate.foreach(IteratorTemplate.scala:32)
    at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
    at kafka.consumer.KafkaStream.foreach(KafkaStream.scala:25)
    at com.twitter.zipkin.receiver.kafka.KafkaStreamProcessor.run(KafkaStreamProcessor.scala:17)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
```

**Comment by codefromthecrypt** [comment]
The issue you are getting is because you are running an older version
of cassandra. Please update per below:
https://groups.google.com/d/msg/zipkin-user/s6NZctGdNJM/VpytyTsICQAJ

FYI The scala code is no longer supported, and the zipkin-java code
that replaces it has a workaround to this issue. Shortly this code
will replace the master branch per #1119

**Comment by codefromthecrypt** [comment]
closing as the current code doesn't throw this exception

**Comment by codefromthecrypt** [comment]
note if you are running an unpatched, or not recently patched version of cassandra 2.1 you'll want to set `CASSANDRA_ENSURE_SCHEMA=false` until below. For example, I've adhoc-tested 2.1.14 and that works in degraded mode, but 2.1.1 does not.

https://groups.google.com/forum/#!topic/zipkin-user/s6NZctGdNJM
